### PR TITLE
Feature/task 891 multiline state labels

### PR DESCRIPTION
**Pull request by franzTobiasDLR**
Make state labels mutli line....

'MY_STATE' then looks like this:
MY_STATE

'TestStateBla' like this:
TestStateBla

(Also of course breaking on white space)

Closing #891

### CODE REVIEW

File: de.dlr.sc.virsat.model.extension.statemachines.ui/src/de/dlr/sc/virsat/model/extension/statemachines/ui/diagram/features/label/MultilineLabelFormatter.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+		StringBuilder builder = new StringBuilder();
+		originalLabel = originalLabel.replace('_', ' ');
+		char[] charArray = originalLabel.toCharArray();
+        for (int i = 0; i < charArray.length; i++) {
+        	if (Character.isUpperCase(charArray[i]) && i > 0 && !Character.isUpperCase(charArray[i - 1])) {
+            	builder.append(' ');
+            }
+            builder.append(charArray[i]);
+        }
+		return WordUtils.wrap(builder.toString(), 1);
```
Comment (SaMuellerDLR) [inline de.dlr.sc.virsat.model.extension.statemachines.ui/src/de/dlr/sc/virsat/model/extension/statemachines/ui/diagram/features/label/MultilineLabelFormatter.java:36]: Maybe its just github but the whitespacing seems off in this method.

File: de.dlr.sc.virsat.model.extension.statemachines.ui/src/de/dlr/sc/virsat/model/extension/statemachines/ui/diagram/features/label/MultilineLabelFormatter.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+	 * Formating a multi line label so that it wraps if following the camel case convention
+	 * @param originalLabel the original label
+	 * @return the formatted label
+	 */
+	public String getLabel(String originalLabel) {
+		StringBuilder builder = new StringBuilder();
+		originalLabel = originalLabel.replace('_', ' ');
+		char[] charArray = originalLabel.toCharArray();
+        for (int i = 0; i < charArray.length; i++) {
+        	if (Character.isUpperCase(charArray[i]) && i > 0 && !Character.isUpperCase(charArray[i - 1])) {
```
Comment (SaMuellerDLR) [inline de.dlr.sc.virsat.model.extension.statemachines.ui/src/de/dlr/sc/virsat/model/extension/statemachines/ui/diagram/features/label/MultilineLabelFormatter.java:31]: Could we start the for loop with i = 1 and then remove this condition?

File: de.dlr.sc.virsat.model.extension.statemachines.ui/src/de/dlr/sc/virsat/model/extension/statemachines/ui/diagram/features/label/MultilineLabelFormatter.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+ *******************************************************************************/
+package de.dlr.sc.virsat.model.extension.statemachines.ui.diagram.features.label;
+
+import org.apache.commons.lang.WordUtils;
+
+/**
+ * Formating a multi line label
+ *
+ */
+public class MultilineLabelFormatter {
```
Comment (SaMuellerDLR) [inline de.dlr.sc.virsat.model.extension.statemachines.ui/src/de/dlr/sc/virsat/model/extension/statemachines/ui/diagram/features/label/MultilineLabelFormatter.java:18]: Lets move the formatter to the model plugin, then we can also test it.
I dont see any UI dependencies.

File: de.dlr.sc.virsat.model.extension.statemachines.ui/src/de/dlr/sc/virsat/model/extension/statemachines/ui/diagram/features/label/MultilineLabelFormatter.java (hunk 1 above)
Comment (SaMuellerDLR) [inline de.dlr.sc.virsat.model.extension.statemachines.ui/src/de/dlr/sc/virsat/model/extension/statemachines/ui/diagram/features/label/MultilineLabelFormatter.java:36]: Can we define a constant for the second parameter?

**Review by SaMuellerDLR — COMMENTED** [review]
Cool idea and easy solution.

**Review by GeorgeNasrDLR — APPROVED** [review]
(y) nice

**Comment by SaMuellerDLR** [comment]
@franzTobiasDLR Ups that was supposed to be a request changes.

**Review by SaMuellerDLR — APPROVED** [review]
Great!

**Pull request merged**